py/makeqstrdefs.py

```diff
@@ -39,11 +40,26 @@ def preprocess():
         os.makedirs(os.path.dirname(args.output[0]))
     except OSError:
         pass
-    with open(args.output[0], "w") as out_file:
-        if csources:
-            subprocess.check_call(args.pp + args.cflags + csources, stdout=out_file)
-        if cxxsources:
-            subprocess.check_call(args.pp + args.cxxflags + cxxsources, stdout=out_file)
+
+    def pp(flags):
+        def run(files):
+            return subprocess.check_output(args.pp + flags + files)
+
+        return run
+
+    cpus = multiprocessing.cpu_count()
+    p = multiprocessing.dummy.Pool(cpus)
+    with open(args.output[0], "wb") as out_file:
+        for flags, sources in (
+            (args.cflags, csources),
+            (args.cxxflags, cxxsources),
+        ):
+            if not sources:
```

Comment:
Shouldn't be needed? chunks will be empty as well if source is empty and I assume `imap` will also not do anything if there's no input

Reply:
@stinos when `sources` is empty, `batch_size` is zero, and you can't do `range(x, y, 0)`.

I could solve this by making a minimum batch size of 1 or something but the `if not sources:` seems clearer to me.

Reply:
For me it's the opposite actually :) looking at the line calculating batch_size I really can't be bothered figuring out what it does for each possible number, so I see a calculation which for all i know could return 0, which makes me wondering what's going to happen in that case.